### ISSUE TITLE
Releasing 2.0.0

### ISSUE THREAD

**Issue by thomasbird** [LeapBeyond/scrubadub#91]
I'm thinking about releasing v2.0.0 soon, but was keen to get any user feedback on the latest changes. 

If anyone has feedback on the API changes, new documentation or code, please let me know!

The latest docs are available at:
https://scrubadub.readthedocs.io/en/develop/index.html

Tagging a few recent contributors: @aCampello @deanmalmgren @mirandachong

**Comment by aCampello** [comment]
Nice! I've seen that there is been a lot of recent activity. I'll test them and get bak to you! :)

**Comment by deanmalmgren** [comment]
@thomasbird I've been lurking from afar and I love the direction it's going. 👏 👏 👏 

If I could offer one suggestion it would be to make `spacy` the default name detector on installation, with the optionality to use `textblob` (although why you would now that `spacy` is working, I don't know) or `stanford`. I don't know when `spacy` v3 is being properly released, but maybe the nightly builds are good enough for now to make it a reasonable thing to do for `scrubadub`. 

This is all really great work though and I deeply appreciate the effort and thoughtfulness you all have put into this. Thank you!

**Comment by thomasbird** [comment]
@deanmalmgren Yeah I agree spacy is a good default detector for names, however it is slow and that's my main reason for not activating it by default. But I think you have a good point, it'd be great for users to be able to use it out of the box. Using the `./tests/benchmark_time.py` script it takes 0.01s without spacy for a scrubber.clean, 1s with the small spacy model and 25s with the new spacy transformer model.

Also making the whole of scrubadub dependant on an unstable spacy makes me nervous (at the moment it's just an extra `pip install scrubadub[spacy]`). Currently when you `pip install spacy-nightly` in a blank environment you get a dependency issue with the latest version of thinc (also from the developers of spacy), which I wouldn't want to force on our users.

There's also complexity around how spacy installs models as packages that aren't on pypi. In a fresh install if we include spacy as default scrubadub will break as there will be no models installed. But I think this can be improved with code.

TL;DR, Maybe we could do this once spacy v3 is stable, but I wouldn't want to do it now. However, maybe we could hold out for the next release until v3 is stable and make it default then? Thoughts?

@aCampello Thanks! :smile:

**Comment by deanmalmgren** [comment]
Ahhhh... I didn't catch that spacy was slow relative to textblob when I read through the docs. I wonder if it would be appropriate to call out that tradeoff in the [accuracy table](https://scrubadub.readthedocs.io/en/latest/accuracy.html#pseudo-data-performance) to help potential users understand that tradeoff?

Making spacy default once v3 is stable certainly sounds like a reasonable approach to me. Thanks for sharing your reflections here.

**Comment by aCampello** [comment]
Just calling out the fact that Spacy v3 is now stable (https://github.com/explosion/spaCy/releases) and not on nightly anymore! We can still keep it as an extra, but perhaps update the requirements to use the stable version.

**Comment by adilosa** [comment]
Hey @thomasbird, v2 looks great - any chance it will be released to pypi soon?

**Comment by thomasbird** [comment]
Just to follow up with the discussion here. 

TL;DR scrubadub v2.0.0 has been released.

I didn't like how we were setting up scrubadub previously to depend on many projects. With the addition of the spacy/stanford/... detectors scrubadub was depending on lots of packages. Also working with some people using scrubadub, some of them didn't really get using extras to install things, using seperate packages seemed to be more understandable. So I broke scrubadub up into some smaller packages (scrubadub, scrubadub_address, scrubadub_spacy and scrubadub_stanford) and changed the detector registration to allow the detectors to be in separate packages. Hopefully this also is a little more future proof than the previous rc too.

Now that that's all done, it's on pypi. Now I'm waiting for the first bugs to come in.
